### PR TITLE
Mingw

### PR DESCRIPTION
**Pull request by KayEss**
This implements some of what is needed to support Windows, at least, using cross compiling or native compiling with the mingw-w64 tools.

### CODE REVIEW

File: src/poll.hpp [hunk 1]
```diff
@@ -67,5 +67,28 @@ namespace felspar::io {
...
+        return WSAGetLastError();
+#else
+        return errno;
+#endif
+    }
+    inline bool would_block(auto const err) {
+#ifdef FELSPAR_WINSOCK2
+        return err == WSAEWOULDBLOCK;
+#else
+        return err == EAGAIN or err == EWOULDBLOCK;
```
Comment (jtv) [inline src/poll.hpp:82]: `EAGAIN` doesn't really sound like a blocking condition...  Perhaps this function doesn't answer the question of "would this block?" so much as "do I need to continue waiting?"

File: src/poll.iops.cpp [hunk 2]
```diff
@@ -62,92 +68,117 @@ struct felspar::io::poll_warden::write_some_completion :
...
     socklen_t addrlen;
     void cancel_iop() override { std::erase(self->requests[fd].writes, this); }
     felspar::coro::coroutine_handle<>
             await_suspend(felspar::coro::coroutine_handle<> h) override {
         handle = h;
+#ifdef FELSPAR_WINSOCK2
+        if (auto err = ::connect(fd, addr, addrlen); err != SOCKET_ERROR) {
+            return handle;
+        } else if (auto const wsae = WSAGetLastError();
+                   wsae == WSAEWOULDBLOCK) {
```
Comment (jtv) [inline src/poll.iops.cpp]: Not so easy to scroll back on my phone but didn't you introduce a "would block" helper earlier?

**Review by jtv — APPROVED** [review]
Plenty of stuff in there of which I don't know the meaning, but the abstraction of the socket stuff looks good to me.

File: src/poll.hpp (hunk 1 above)
Comment (KayEss) [reply src/poll.hpp:82]: From the [write man page](https://man7.org/linux/man-pages/man2/write.2.html)

       EAGAIN or EWOULDBLOCK
              The file descriptor fd refers to a socket and has been
              marked nonblocking (O_NONBLOCK), and the write would
              block.  POSIX.1-2001 allows either error to be returned
              for this case, and does not require these constants to
              have the same value, so a portable application should
              check for both possibilities.

File: src/poll.iops.cpp (hunk 2 above)
Comment (KayEss) [reply src/poll.iops.cpp]: There is, but this is Windows code anyway. Still, I guess no harm in using it

File: src/poll.hpp (hunk 1 above)
Comment (jtv) [reply src/poll.hpp:82]: My apologies.  I was thinking of `EINTR` for some reason.

**Pull request merged**